src/render/index.js

```diff
@@ -93,19 +117,18 @@ export const renderFromHTML = R.curry((el, html) =>
                 /**
                  * If it is a container, we're going to do our own updating
                  * and tell morphdom to move on.
+                 *
+                 * If the container has changed, swap element ourselves.
+                 * This is similar to how React handles it: If a subtree
+                 * is a different component, it just prunes and replaces,
+                 * since the subtree could be different in myriad different
+                 * ways and a full diff would be computationally
+                 * expensive. Additionally, this allows the MutationObserver
+                 * to continue to only worry about add/remove operations
+                 * instead of attribute mutations.
                  */
-                const containerKey = fromEl.getAttribute(CONTAINER_ATTRIBUTE);
-
-                // If the container has changed, swap element ourselves.
-                // This is similar to how React handles it: If a subtree
-                // is a different component, it just prunes and replaces,
-                // since the subtree could be different in myriad different
-                // ways and a full diff would be computationally
-                // expensive. Additionally, this allows the
-                // MutationObserver to continue to only worry about
-                // add/remove operations instead of attribute mutations.
-                if (containerKey !== toEl.getAttribute(CONTAINER_ATTRIBUTE) ||
-                    el.getAttribute(KEY_ATTRIBUTE) !== toEl.getAttribute(KEY_ATTRIBUTE)
+                if (fromEl.getAttribute(CONTAINER_ATTRIBUTE) !== toEl.getAttribute(CONTAINER_ATTRIBUTE) ||
+                    fromEl.getAttribute(KEY_ATTRIBUTE) !== toEl.getAttribute(KEY_ATTRIBUTE)
```

Comment:
For anyone reading along, this was the cause of the bug. This should compare to fromEl. It also provided an opportunity to clean up the rendering logic overall, since we don't need to compare the template to the actual element in production.